pkg/proxy/integrations/httpparser/httpparser.go

```diff
@@ -110,122 +110,183 @@ func ProcessOutgoingHttp(request []byte, clientConn, destConn net.Conn, h *hooks
 }
 
 // Handled chunked requests when content-length is given.
-func contentLengthRequest(finalReq *[]byte, clientConn, destConn net.Conn, logger *zap.Logger, contentLength int) {
+func contentLengthRequest(finalReq *[]byte, clientConn, destConn net.Conn, logger *zap.Logger, contentLength int) error {
 	for contentLength > 0 {
 		clientConn.SetReadDeadline(time.Now().Add(5 * time.Second))
 		requestChunked, err := util.ReadBytes(clientConn)
 		if err != nil {
 			if err == io.EOF {
-				logger.Error("connection closed by the user client", zap.Error(err))
-				break
+				logger.Error("connection closed by the user client")
+				return err
 			} else if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
 				logger.Info("Stopped getting data from the connection", zap.Error(err))
 				break
 			} else {
-				logger.Error("failed to read the response message from the destination server", zap.Error(err))
-				return
+				logger.Error("failed to read the response message from the destination server")
+				return err
 			}
 		}
 		logger.Debug("This is a chunk of request[content-length]: " + string(requestChunked))
 		*finalReq = append(*finalReq, requestChunked...)
 		contentLength -= len(requestChunked)
-		_, err = destConn.Write(requestChunked)
-		if err != nil {
-			logger.Error("failed to write request message to the destination server", zap.Error(err))
-			return
+
+		// destConn is nil in case of test mode.
```

Comment:
How was this working before, Like did we used to get this error `logger.Error("failed to write request message to the destination server", zap.Error(err))` before in test mode..?

Reply:
We might have not encountered this chunking during test mode before, otherwise, it would have panicked

Reply:
got it